framework/src/main/java/org/checkerframework/common/reflection/DefaultReflectionResolver.java

```diff
@@ -517,7 +521,7 @@ private List<Symbol> getMethodSymbolsfor(
     List<Symbol> result = new ArrayList<>();
     ClassSymbol classSym = (ClassSymbol) sym;
     while (classSym != null) {
-      for (Symbol s : classSym.getEnclosedElements()) {
+      for (Symbol s : sym.getEnclosedElements()) {
```

Comment:
It looks like `classSym` gets updated on each loop iteration? Not sure this will behave the same as the previous code.